content/news/2021/2021-09-10-how-to-supplement-cloudwatch-and-save-your-sanity.md

```diff
@@ -0,0 +1,171 @@
+
+With this basic function in place, you can deploy your Lambda with Claudia by running: 
+
+```
+claudia create --region us-east-1 --handler lambda.handler --profile claudia
+```
+
+You can, of course, choose a different region. If you didn't use `claudia` for your AWS credentials profile name, be sure to specify the name you did use. Once this command is done, you can test your Lambda with: 
+
+```
+claudia test-lambda --profile claudia
+```
+
+This should simply print out the "hello world" statement from your Lambda. This is a good start. Next, we'll update our function to include some basic logging and see how that looks in Cloudwatch. 
+
+## Logging With Cloudwatch
+
+We have a basic Lambda, but let's make it so we can see some info in Cloudwatch. Update your `lambda.js` file to look like this: 
+
+```js
+exports.handler = function (event, context, callback) {
+  try {
+    console.log("Firing up the lambda...");
+    console.log(event);
+    console.log("Success!");
+    context.succeed('hello world'); 
+  } catch (error) {
+    console.log(error);
+    callback(Error(e))
+  }	
+};
+```
+
+To update your Lambda function, run: 
+
+```
+claudia update --profile claudia
+```
+
+Once that is updated, you can test your function again by running: 
+
+```
+claudia test-lambda --profile claudia
+```
+
+After that successfully runs, you can now go into and see the logs. You'll want to click on Log Groups on the left side: 
+
+![Log Groups Example](./Cloudwatch_Log_Groups.png)
+
+Then you will logs grouped by area in which they originated. You should see your lambda in the list. Click on it and then you will be taken to log streams. The most recent are at the top. If you click the top stream, you should see events for your Lambda that can be expanded. 
+
+![Log Streams Example](./Lamba_Log_Stream.png)
+
+You can see with each event expanded that the console log statements are printed in Cloudwatch. This is useful for ad hoc checking on things, but if we want to be alerted when something happens, we may want to combine logging with error or event handling. Let's do that now!
+
+## Connecting Exceptionless 
+
+The first thing to do is sign up for your free [Exceptionless](https://exceptionless.com) account if you haven't done so. Once you have signed up, you can create an API key by going to your project settings page: 
+
+![Project settings](./APIKeys.png)
+
+Save your API Key because you'll need it here soon. We need to now install the new [Exceptionless JavaScript client](2021-09-09-announcing-the-new-exceptionless-javascript-client.md) into our Lambda project. To do this, return to your Lambda project directory and run the following: 
+
+```
+npm i @exceptionless/node
+```
+
+Once that's installed, update your Lambda code to look like this: 
+
+```js
+const { Exceptionless } = require("@exceptionless/node");
```

Comment:
This works? I didn't think requires works with ESM. Another con is we say our library is ESM and we have nothing about any other module format in our documentation. Is this opening a can of worms?

Reply:
Whoops forgot to update that. Def doesn't work.

Reply:
Oh this is on a post that is not going live.I'm going to delete that one because I have it on another branch